catalyst_voices/packages/catalyst_voices_brands/lib/src/theme_extensions/brand_assets.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:flutter/material.dart';
+
+/// A `ThemeExtension` that holds brand-specific assets for theming purposes.
+///
+/// `BrandAssets` is used to encapsulate theme-dependent assets such as the
+/// brand's logo and logo icon as SVG images which can be utilized throughout
+/// the application wherever is required.
+///
+/// Example usage:
+/// ```dart
+/// final logo = Theme.of(context).brandAssets.logo;
+/// final logoIcon = Theme.of(context).brandAssets.logoIcon;
+/// ```
+@immutable
+class BrandAssets extends ThemeExtension<BrandAssets> {
+  final SvgGenImage logo;
+  final SvgGenImage logoIcon;
+
+  const BrandAssets({
+    required this.logo,
+    required this.logoIcon,
+  });
+
+  @override
+  ThemeExtension<BrandAssets> copyWith({
+    SvgGenImage? logo,
+    SvgGenImage? logoIcon,
+  }) {
+    return BrandAssets(
+      logo: logo ?? this.logo,
+      logoIcon: logo ?? this.logoIcon,
+    );
+  }
+
+  @override
+  BrandAssets lerp(
+    ThemeExtension<BrandAssets>? other,
+    double t,
+  ) {
+    if (other is! BrandAssets) {
```

Comment:
I've no idea honestly how to implement lerp correctly. The method is using during animations, i.e. when changing from one theme to another (light to dark, etc) but to animate the images - too much work probably.

I kinda guess that we should do something like:

```
if (other is! BrandAssets) return this;
else if (t >= 0.5) return other;
else return this;
```

This way while we are halway through the animation we already show the other state, instead of showing the original state for the whole time. But you can do more research about that if you like ;)

Reply:
Followed your advice and implemented what you suggested.
I did some research about using opacity during the animation but it looked too complex for the use case we have right now.